ape_etherscan/explorer.py

```diff
@@ -1,14 +1,32 @@
 from ape.api import ExplorerAPI
+from ape.types import ABI, ContractType
+from typing import List
+import requests
+import json
 
 ETHERSCAN_URI = (
     lambda n: (f"https://{n}.etherscan.io/" if n != "mainnet" else "https://etherscan.io/")
     + "{0}/{1}/"
 )
 
+ETHERSCAN_API_URI = lambda n: (
+    f"https://api-{n}.etherscan.io/api" if n != "mainnet" else "https://api.etherscan.io/api"
+)
+
 
 class Etherscan(ExplorerAPI):
     def get_address_url(self, address: str) -> str:
         return ETHERSCAN_URI(self.network.name).format("address", address)
 
     def get_transaction_url(self, transaction_hash: str) -> str:
         return ETHERSCAN_URI(self.network.name).format("tx", transaction_hash)
+
+    def get_contract_abi(self, address: str) -> List[ABI]:
+        response = requests.get(
+            ETHERSCAN_API_URI(self.network.name),
+            params={"module": "contract", "action": "getabi", "address": address},
+        )
+        abi_string = response.json()["result"]
```

Comment:
An improvement here might be to do

```suggestion
        abi_string = response.json().get("result")
```

And then afterward, raise a custom exception stating that it was not able to retrieve the ABI. Otherwise, the user will see an error like `KeyError: "result"` and that won't mean much to them

Reply:
https://github.com/ApeWorX/ape-etherscan/issues/6

Reply:
I think i've updated this in my new push https://github.com/ApeWorX/ape-etherscan/pull/2/commits/84eb9dd2f4ed134245514851ebf3e35841ba7db5